src/tests/test_serialization.cpp

```diff
@@ -134,4 +135,50 @@ BOOST_AUTO_TEST_CASE( test_wildcard_string_match )
 	BOOST_CHECK(!utils::wildcard_string_match("", "???"));
 }
 
+BOOST_AUTO_TEST_CASE( test_base64_encodings )
+{
+	const std::vector<uint8_t> empty;
+	const std::string empty_b64;
+	const std::string empty_c64;
+	const std::vector<uint8_t> foo = {'f', 'o', 'o'};
+	const std::string foo_b64 = "Zm9v";
+	const std::string foo_c64 = "axqP";
+	const std::vector<uint8_t> foob = {'f', 'o', 'o', 'b'};
+	const std::string foob_b64 = "Zm9vYg==";
+	const std::string foob_c64 = "axqPW/";
+
+	std::vector<uint8_t> many_bytes;
+
+	many_bytes.reserve(1024);
```

Comment:
This should probably be `resize` rather than `reserve`, since you're not using `push_back` in the loop body.

Reply:
Fixed.